### DISCUSSION TITLE
0005 XLS-5.1d NetworkID extension to X-Addresses

### DISCUSSION THREAD

**Discussion by RichardAH** [XRPLF/XRPL-Standards#142] [XLS Proposals]
## The earlier version can be found by looking at the edit history.

**Comment by mvadari** [comment]
Why remove the X vs T  prefix? I like that I can tell at-a-glance if this account holds real funds or fake test funds.

**Reply by WietseWind** [reply]
@RichardAH I agree with @mvadari - let's keep the prefix? Makes it easy, optically, at first sight if we're dealing with real or fake.

**Reply by sublimator** [reply]
Do XRPL main and XRPL test have different network ids?

**Reply by WietseWind** [reply]
Main 0
Test 1

**Reply by sublimator** [reply]
If they do, any ux/security concerns? 

<img width="601" alt="image" src="https://github.com/XRPLF/XRPL-Standards/assets/525211/ada6c4fb-8486-4a6a-95f6-f62c5654c67a">

Hard to tell what this means, given it's using a string, but ?

**Reply by sublimator** [reply]
Jinx! haha

**Reply by sublimator** [reply]
So how do you statelessly determine T or X for any given network, uhm, family?

**Reply by sublimator** [reply]
I mean beyond the obvious "just encode with an extra byte"

>  any ux/security concerns?

All that's over my head

**Reply by WietseWind** [reply]
Ecosystem standard, keeping track of a list.

Alternatively, providing most networks are testnets, we could do something like this:

- 0 (mainnet) % 19 = 0 = mainnet
- 1 (testnet) % 19 = 1 = testnet
- 21337 (Xahau mainnet) % 19 = 0 = mainnet
- 21338 (Xahau testnet) % 19 = 1 = testnet

**Reply by sublimator** [reply]
Modulo style would still require a list/reference point though, right?

**Reply by sublimator** [reply]
Oh, wait! I see

**Reply by sublimator** [reply]
In any case, over my head, but I can see why maybe the prefix was removed if main and test have different ids

**Reply by WietseWind** [reply]
Exactly. With one big difference: X/T address prefix would allow end users to visually identify live/test from the address they are about to use.

**Reply by sublimator** [reply]
Definitely see the value of that in principle - but ideally can be sure the visuals match what's actually encoded

**Reply by sublimator** [reply]
>>but ideally can be sure the visuals match what's actually encoded

Hrmmmmmmmm

**Reply by sublimator** [reply]
You all are planning to support addresses that work across networks eh? Is that really wise? I mean you have NetworkID/Sequence transactor level protections but still ?

Without a clear disambiguator for ANY network family at the front, what's the point of disambiguating the class of network?

Cause you're going to need to show the network in the UI somehow anyway? Assuming you support multiple XRP based chains?

It's a question of "does the sending network id (surely shown and/or selected in the UI) match the recipient network id ?"

**Reply by WietseWind** [reply]
@sublimator Network should either be visible in the UI, indeed, or the app & audience completely abstracts it away for the end user & the app would know what context to use based on the address with the network being encoded. In those cases the address will probably even be abstracted away anyway using any of the million naming/resolve schemes. 

Crypto context: app switches to the right network & offers the right context based on the address. 

Retail context: app takes care of everything. Address most likely passed with some human readable / QR / BLE / ... & then resolved / ... - if app supports and moves on with a retail friendly flow. It's not unlikely a certain app+end user interaction touches multiple networks. A destination address in those cases signals where to deliver, not even specifically which network it has to come from.

**Reply by sublimator** [reply]
I think the only thing I can say without thinking about it more (and playing with some UIs) is that it seems somewhat pointless to have "human readable" prefixes unless they are meaningful and can be trusted? Ideally the prefix would visibly encode the network id. Devil's in the details though.

> not even specifically which network it has to come from.

Interesting

**Comment by MarvinSunday** [comment]
It's a good and it will help avoid sending funds to wrong network now that the XRPL Community is expanding.
Great work!

**Comment by mvadari** [comment]
@RichardAH should this spec be moved into PR and kept in the repo or should it be closed? Note that this does not mean that the spec has been accepted/implemented/finalized.

I will move it to PR by copy-pasting the existing draft if there is no response in the next week.
